developing.rst

```diff
@@ -18,9 +26,65 @@ Generate Code from protobuf
 
      make download_tooling
 
+Generate Docs from protobuf
+----------------------------
+
 #. To add new dependencies for a doc, modify ``doc_requirements.in`` and then
 
    .. prompt:: bash
 
      make doc-requirements.txt
 
+Docs structure
+**************
+
+The index.rst files for protos are kept in parallel folder structure under the docs folder.
+All the proto definitions are within protos/flyteidl and there corresponding docs are kept in protos/docs
+
+Each module in protos has same named module under the docs also.
+eg : protos/flyteidl/core has same named doc structure placing it index and other documentation files in protos/docs/core
+
+
+Docs Generation
+***************
+
+Relies on protoc plugin pseudomuto/protoc-gen-doc for generating the docs.
+Currently pseudomuto/protoc-gen-doc is being used for generating the RST files which are placed in protos/docs and corresponding folder based on the module.
+Until this code is checkedin https://github.com/pseudomuto/protoc-gen-doc/pull/440, use the branch to build the protoc-gen-doc executable
+Follow the steps to build and use the protoc-gen-doc
+
+#. Clone the git repo protoc-gen-doc
+
+   .. prompt:: bash
+
+     git clone https://github.com/paweld2/protoc-gen-doc.git
```

Comment:
Why do we need to do this? We can just reference the gitsha in go get right? Other option is to fork the repo into flyteorg and use the fork?

Reply:
Or branch

Reply:
Done. Using boiler plate code now and updated the documentation. 
Also currently the gihub workflows don't call make generate and hence don't find any issues. 
We will have to fix the github workflows to use it so that the DELTA_CHECK catches any new files